pkg/config/engine/config.go

```diff
@@ -0,0 +1,85 @@
+/**
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package engine
+
+// A Config represents a config for a container engine.
+// These include container, cri-o, and docker.
+// The config is logically split into a Source and Destination. This allows an
+// existing config to be updated (Source == Destination) or runtime-specific
+// settings to be written to a new config. The latter is useful when creation
+// NVIDIA-specific drop-in files for container engines that support this.
+type Config struct {
+	Source      RuntimeConfigSource
+	Destination RuntimeConfigDestination
+}
+
+// A RuntimeConfigSource allows runtime-specific settings to be READ from a
+// config.
+type RuntimeConfigSource interface {
+	DefaultRuntime() string
+	GetRuntimeConfig(string) (RuntimeConfig, error)
+	GetDefaultRuntimeOptions() interface{}
+	String() string
+}
+
+// A RuntimeConfigDestination allows a runtime with specific settings to be
+// WRITTEN to a config.
+type RuntimeConfigDestination interface {
+	AddRuntimeWithOptions(string, string, bool, interface{}) error
+	EnableCDI()
+	RemoveRuntime(string) error
+	Save(string) (int64, error)
+	String() string
+}
+
+// AddRuntime adds a runtime to the destination config and optionally sets it as the default.
+// The options to apply to the added runtime are read from the source config
+// default runtime.
+func (c *Config) AddRuntime(name string, path string, setAsDefault bool) error {
+	options := c.Source.GetDefaultRuntimeOptions()
+	return c.Destination.AddRuntimeWithOptions(name, path, setAsDefault, options)
+}
```

Comment:
I believe this means the changes I made in https://github.com/NVIDIA/nvidia-container-toolkit/pull/1256 will no longer take affect. If `default_runtime_name=nvidia` is set in the top-level config, and `setAsDefault=false`, we don't actually unset `nvidia` as the default runtime.